src/parse.js

```diff
@@ -102,9 +102,17 @@ export default function parse(html, options) {
         // calculate correct end of the content slice in case there's
         // no tag after the text node.
         const end = html.indexOf('<', start)
-        const content = html.slice(start, end === -1 ? undefined : end)
-        // if a node is nothing but whitespace, no need to add it.
-        if (!/^\s*$/.test(content)) {
+        let content = html.slice(start, end === -1 ? undefined : end)
+        // if a node is nothing but whitespace, collapse it as the spec states:
+        // https://www.w3.org/TR/html4/struct/text.html#h-9.1
+        if (/^\s*$/.test(content)) {
+          content = ' ';
+        }
+        // don't add whitespace-only text nodes if they would be trailing text nodes
+        // or if they would be leading whitespace-only text nodes:
+        //  * end > -1 indicates this is not a trailing text node
+        //  * leading node is when level is -1 and parent has length 0
+        if ((options.respectWhitespace && end > -1 && level + parent.length >= 0) || content !== ' ') {
```

Comment:
I'm trying to figure out if there's a scenario where you would not want `respectWhitespace` to be true. 

I'd rather just change behavior than add an option, unless we can think of a good reason why anyone would ever want it to be false.

Reply:
I was thinking the same when creating the PR, but did not want to break the existing behaviour.
If you want to create a major and always respect the whitespaces it's even better ;-)